test/test_add_mode.py

```diff
@@ -13,33 +13,32 @@
 from ffcx.naming import cdtype_to_numpy, scalar_to_value_type
 
 
-@pytest.mark.parametrize("mode",
-                         [
-                             "double",
-                             "float",
-                             "long double",
-                             "double _Complex",
-                             "float _Complex"
-                         ])
+@pytest.mark.parametrize(
+    "mode", ["double", "float", "long double", "double _Complex", "float _Complex"]
+)
 def test_additive_facet_integral(mode, compile_args):
     element = basix.ufl.element("Lagrange", "triangle", 1)
     u, v = ufl.TrialFunction(element), ufl.TestFunction(element)
     a = ufl.inner(u, v) * ufl.ds
     forms = [a]
     compiled_forms, module, code = ffcx.codegeneration.jit.compile_forms(
-        forms, options={'scalar_type': mode}, cffi_extra_compile_args=compile_args)
+        forms, options={"scalar_type": mode}, cffi_extra_compile_args=compile_args
+    )
 
     for f, compiled_f in zip(forms, compiled_forms):
         assert compiled_f.rank == len(f.arguments())
 
     ffi = module.ffi
     form0 = compiled_forms[0]
 
-    assert form0.num_integrals(module.lib.exterior_facet) == 1
-    ids = form0.integral_ids(module.lib.exterior_facet)
-    assert ids[0] == -1
+    integral_offsets = form0.form_integral_offsets
+    ex = module.lib.exterior_facet
+    num_integrals = integral_offsets[ex + 1] - integral_offsets[ex]
+    assert num_integrals == 1
+    integral_id = form0.form_integral_ids[integral_offsets[ex]]
+    assert integral_id == -1
 
-    default_integral = form0.integrals(module.lib.exterior_facet)[0]
+    default_integral = form0.form_integrals[integral_offsets[ex]]
```

Comment:
This here is implicitly tied to the ordering of the enum in DOLFINx, which I'm not sure is a good idea. See: https://github.com/FEniCS/ffcx/pull/589#issuecomment-1676768811

Reply:
The order is explicitly defined in ufcx.h

Reply:
It is still an issue for whenever we generate the offsets (as that is based on strings in a for-loop, not the enum).

Reply:
I have annotated the point where this happens in ffcx.